src/main/java/co/cask/WholeFileSource.java

```diff
@@ -90,13 +90,16 @@ public String getInputFormatClassName() {
   }
 
   @Override
-  public void transform(KeyValue<NullWritable, BytesWritable> input,
+  public void transform(KeyValue<String, BytesWritable> input,
                         Emitter<StructuredRecord> emitter) throws Exception {
-    emitter.emit(StructuredRecord.builder(outputSchema).set("body", input.getValue().getBytes()).build());
+    emitter.emit(StructuredRecord.builder(outputSchema)
+                   .set("fileName", input.getKey())
```

Comment:
Why just the name instead of the full path?

Reply:
using full path now.

Reply:
then you better call this "filePath"?

Reply:
updated it